src/hotspot/share/gc/shenandoah/shenandoahGenerationalHeap.cpp

```diff
@@ -121,19 +121,19 @@ ShenandoahGenerationalHeap::TransferResult ShenandoahGenerationalHeap::balance_g
   shenandoah_assert_heaplocked_or_safepoint();
 
   ShenandoahOldGeneration* old_gen = old_generation();
-  const size_t old_region_surplus = old_gen->get_region_surplus();
-  const size_t old_region_deficit = old_gen->get_region_deficit();
-  old_gen->set_region_surplus(0);
-  old_gen->set_region_deficit(0);
+  const ssize_t old_region_balance = old_gen->get_region_balance();
+  old_gen->set_region_balance(0);
 
-  if (old_region_surplus) {
-    bool success = generation_sizer()->transfer_to_young(old_region_surplus);
+  if (old_region_balance > 0) {
+    const auto old_region_surplus = checked_cast<size_t>(old_region_balance);
```

Comment:
Just curious: what happens with a checked_cast<size_t> if old_region_balance < 0.  (I see that's impossible here, but what is the point of declaring checked_cast?)

Reply:
`checked_cast` asserts that the cast did not change the value, defined here: https://github.com/openjdk/shenandoah/blob/master/src/hotspot/share/utilities/checkedCast.hpp#L39. If `old_region_balance` were less than zero, it would assert out. It will also assert out if `old_region_balance` were greater than `SSIZE_MAX`. Which is unlikely, but there is no harm in checking.